server/src/main/java/io/crate/execution/jobs/RootTask.java

```diff
@@ -219,9 +196,14 @@ public void start() throws Throwable {
     }
 
     @Nullable
+    @SuppressWarnings("unchecked")
     public <T extends Task> T getTaskOrNull(int phaseId) {
-        //noinspection unchecked
-        return (T) tasksByPhaseId.get(phaseId);
+        for (Task task : orderedTasks) {
```

Comment:
Although this changes the alorithm from a `O(1)` of the map lookup to a `O(n)`, `n` is usually 2 or 3, so it's likely faster than the `O(1)` of the map was.